core/dive.cpp

```diff
@@ -367,91 +367,117 @@ static bool has_unknown_used_cylinders(const struct dive &dive, const struct div
 	return num > 0;
 }
 
-void per_cylinder_mean_depth(const struct dive *dive, struct divecomputer *dc, int *mean, int *duration)
-{
-	int num_used_cylinders;
-
-	if (dive->cylinders.empty())
-		return;
-
-	for (size_t i = 0; i < dive->cylinders.size(); i++)
```

Comment:
This is not actually functionally equivalent - have you verified that the values assigned to  `mean` and `duration` are not used if there is dive computer?